[Caleb Ng Yi Yong] iP

calebyyy opened the pull request:
# Calebyyy PR 🎉

>  My PR

## Bullet List of Features
- List **tasks**
- Mark tasks
- Persistent Storage

## Numbered List of Changes
1. Added Javadoc
2. Refactored existing methods
3. Updated unit tests

##  **Task List**
- [x] Changed PR

## Usage

Use `javac *.java` to compile my code

## Releases
[Releases](https://github.com/calebyyy/ip/tags)


## Main Class
```Java
public static void main(String[] args) {
        new Calebyyy().start();
    }

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

calebyyy committed:
ignore class files

calebyyy committed:
Implement skeleton greeting

calebyyy committed:
Add echo feature

calebyyy committed:
feature: Add task management functionality to Calebyyy
refactor: Made code modular

calebyyy committed:
refactor to include oop

calebyyy committed:
created mark as done functionality

calebyyy committed:
Add different task types

calebyyy committed:
organise to folders

calebyyy committed:
Add error handling

calebyyy committed:
bug: deadline should not throw exception

calebyyy committed:
Ui testing

calebyyy committed:
Add delete feature

calebyyy committed:
Added enums

calebyyy committed:
Added unit test

calebyyy committed:
Create save feature

calebyyy committed:
Add save feature

calebyyy committed:
Add time and date feature

calebyyy committed:
Add for date time for event too

calebyyy committed:
Added more OOP

calebyyy committed:
Merge branch 'add-gradle-support'

calebyyy committed:
Add JUnit tests

calebyyy committed:
Add javadocs

calebyyy committed:
Add find feature

calebyyy committed:
Add JavaDoc for ListCommand.java

calebyyy committed:
Add JavaDoc for Ui.java

calebyyy committed:
Add JavaDoc for TaskList.java

calebyyy committed:
Add JavaDoc for Storage.java

calebyyy committed:
Add JavaDoc for Todo.java

calebyyy committed:
Fix styling

calebyyy committed:
Add settings

calebyyy committed:
Merge branch 'A-CheckStyle-Branch'

calebyyy committed:
Add GUI

calebyyy committed:
Add calebyyy to window

TheRareFox commented on .vscode/settings.json:
I dont think this file should be inside your git repo

TheRareFox commented on src/main/java/calebyyy/Main.java:
```diff
@@ -0,0 +1,31 @@
+import javafx.stage.Stage;
+
+/**
+ * A GUI for Duke using FXML.
```
Nice use of a GUI for the chatbot

TheRareFox commented on src/main/java/calebyyy/Storage.java:
```diff
@@ -0,0 +1,99 @@
+        if (!file.exists()) {
+            return;
+        }
+
+        try (BufferedReader reader = new BufferedReader(new FileReader(file))) {
+            String line;
+            while ((line = reader.readLine()) != null) {
+                String[] parts = line.split(" \\| ");
+                String type = parts[0];
+                boolean isDone = parts[1].equals("1");
+                String description = parts[2];
+                Task task = null;
+
+                switch (type) {
+                case "T":
+                    task = new Todo(description);
+                    break;
+                case "D":
+                    String by = parts[3];
+                    task = new Deadline(description, by);
+                    break;
+                case "E":
+                    String from = parts[3];
+                    String to = parts[4];
+                    task = new Event(description, from, to);
+                    break;
+                default:
+                    break;
+                }
+
+                if (task != null) {
+                    if (isDone) {
+                        task.markAsDone();
+                    }
+                    tasklist.addTask(task);
+                }
+            }
```
Try to not have a try block that is so long. Wrap the try block in the minimum amount of code possible, otherwise unchecked errors from the other statements like the case statement could be caught accidentally.

TheRareFox commented on src/main/java/calebyyy/Ui.java:
```diff
@@ -0,0 +1,81 @@
+     * @param taskCount
+     */
+    public void addTaskMessage(Task task, int taskCount) {
+        System.out.println("____________________________________________________________");
```
Since you are printing the seperator so many times, could consider putting it into a static string

TheRareFox commented on src/main/java/calebyyy/commands/AddCommand.java:
```diff
@@ -0,0 +1,60 @@
+        String taskDetails = parts[1];
+
+        Task task;
+        if (commandType.equals("todo")) {
```
Could use a switch case too and make use of the enum command types

TheRareFox reviewed (commented):
Overall excellent work, adheres to OOP principles and is very neat. Good job!

calebyyy committed:
Add unit tests for Parser

calebyyy committed:
Add gradle dependencies

Need them for Jar creation

So I added them

To make gradle as streamlined as possible

calebyyy committed:
A-Assertions

calebyyy committed:
Add assertions

calebyyy committed:
Merge branch 'branch-A-Assertions' of https://github.com/calebyyy/ip into branch-A-Assertions

calebyyy committed:
Check code quality

calebyyy committed:
Merge pull request #1 from calebyyy/branch-A-Assertions

Branch a assertions

calebyyy committed:
Merge branch 'master' into branch-A-CodeQuality

calebyyy committed:
Merge pull request #2 from calebyyy/branch-A-CodeQuality

Branch a code quality

calebyyy committed:
Fix styling

calebyyy committed:
Add duplicate task protection

calebyyy committed:
Fix typo

itsme-zeix commented on src/main/java/calebyyy/Main.java:
```diff
@@ -0,0 +1,31 @@
+ */
+public class Main extends Application {
+
+    private Calebyyy duke = new Calebyyy();
```
Can consider renaming the remnants of duke for decreased confusion in the long term!

itsme-zeix commented on src/main/java/calebyyy/Ui.java:
```diff
@@ -0,0 +1,176 @@
+     * Print goodbye message.
+     */
+    public void byeMessage() {
+        System.out.println("____________________________________________________________");
+        System.out.println(" Bye. Hope to see you again soon!");
+        System.out.println("____________________________________________________________");
```
Perhaps you could move the logic into a method that wraps any String in horizontal lines.

itsme-zeix commented on src/test/java/calebyyy/ParserTest.java:
```diff
@@ -0,0 +1,42 @@
+    }
+
+    @Test
+    public void testGetCommand_validCommand() {
```
Do use the suggested naming for tests: `featureUnderTest_testScenario_expectedBehavior()`
e.g. `sortList_emptyList_exceptionThrown()`, `getMember_memberNotFound_nullReturned`

itsme-zeix reviewed (commented):
LGTM! Looks great overall, just some really minor style nitpicks here and there!

calebyyy committed:
Add invalid date exception

calebyyy committed:
Refactor AddCommand

calebyyy committed:
Add exceptions

calebyyy committed:
Refactor Storage.java

calebyyy committed:
Refactor JavaDoc

calebyyy committed:
Refactor check style

calebyyy committed:
Add user guide

calebyyy committed:
Refactor Ui

calebyyy committed:
Add exception for DeleteCommand

calebyyy committed:
Fix bug in delete

calebyyy committed:
Refactor ReadMe

calebyyy committed:
Add detection for duplicate tasks

calebyyy committed:
Refactor detection for duplicate tasks

calebyyy committed:
Add JavaDoc
